site/docs/sql/00-intro.md

```diff
@@ -0,0 +1,141 @@
+---
+title: SQL Introduction
+description: Introduction to Hazelcast Jet SQL features.
+---
+
+Hazelcast Jet SQL service executes distributed SQL statements over
+Hazelcast collections and external data sets.
+
+It allows you to specify Jet jobs using the well-known SQL language.
+
+**Note:** _The service is in beta state. Behavior and API might change in
+future releases. Binary compatibility is not guaranteed between minor or
+patch releases._
+
+## Overview
+
+In the first release Jet SQL supports following features:
+
+- SQL Queries over [Apache Kafka topics](05-kafka-connector.md) and
+[Files (local and remote)](04-files-connector.md)
+- Joining Kafka or file data with local IMaps (enrichment)
+- Filtering and projection using [SQL expressions
+](https://docs.hazelcast.org/docs/latest-dev/manual/html-single/index.html#expressions)  
+- Aggregating data from files using predefined
+[aggregate functions](00a-basic-commands#aggregation-functions)
+- Receiving query results via Jet client (Java) or writing the results
+to an [IMap](03-imap-connector.md) in the Jet cluster
+- Running continuous (streaming) and batch queries, see
+[Job Management](02-job-management.md)
+
+Notably, the following features are currently unsupported: joins
+arbitrary external data sources, window aggregation, JDBC. We plan to
+support these in the future.
+
+## Example: How to query Kafka using SQL
+
+Consider that we have a topic called `trades` that contain ticker
+updates. Trades are encoded as JSON messages:
+
+```json
+{ key: ticker, value: price }
+```
+
+To use a remote topic as a table in Jet, an EXTERNAL MAPPING must be
+created first. This maps the JSON messages to a fixed list of columns
+with data types:
+
+```java
+// TODO
+CREATE MAPPING trade_topic (
+    __key VARCHAR,
+    field1 INT)
+TYPE IMap
+OPTIONS (
+    keyFormat 'json',
+    valueFormat 'json')
+```
+
+Jet SQL queries can now use the `trade_topic` as a table:
+
+```java
+JetInstance inst = ...;
+try (SqlResult result = inst.getSql().execute("SELECT ...")) {
+    for (SqlRow row : result) {
+        // Process the row.
+    }
+}
+
+```
+
+The query now runs in the Jet cluster and streams results to the Jet
```

Comment:
Since the streaming query possibly never ends, the example above is an infinite loop in the general case.  Maybe it makes sense to rewrite it a bit, to demonstrate how to finish the execution of the infinite query?